### ISSUE TITLE
Add mode switcher & dark DocFx theme

### ISSUE THREAD

**Issue by perlun** [perlang-org/perlang#174]
https://github.com/steffen-wilke/darkfx/issues/1 should have the info required to implement this.

**Comment by perlun** [comment]
This has been mostly accomplished as of #175 being merged yesterday, but there are still some issues to sort out before we can consider this closed:

- The `.js` improvements (to detect browser preferences) should be upstreamed to https://github.com/perlun/darkerfx properly.
- A UI glitch in the show/hide TOC can be seen on mobile when dark mode is enabled (screenshot below)
- There is currently no way to access the "theme switch" on mobile devices. This should be fixed. We can check how docs.microsoft.com has done it.

![image](https://user-images.githubusercontent.com/630613/113124450-32917600-921e-11eb-9960-90b9591ea5bb.png)

**Comment by perlun** [comment]
> The `.js` improvements (to detect browser preferences) should be upstreamed to https://github.com/perlun/darkerfx properly.

Fixed in https://github.com/perlun/darkerfx/commit/5bd8d973265bd42906cdedc4afb61c1304f17e88

> A UI glitch in the show/hide TOC can be seen on mobile when dark mode is enabled (screenshot below)

This bug is btw also present in `darkfx`, so we could at the very least submit an issue to them about it at some point. Not yet fixed but I've started looking into it.

I also found another horribly looking bug, only visible on mobile:

![image](https://user-images.githubusercontent.com/630613/113282396-84560100-92ef-11eb-9654-d47b37c425d2.png)

**Comment by perlun** [comment]
> I also found another horribly looking bug, only visible on mobile. If you click Show / Hide Table of Contents one or more times, you can run into this:

Not reproducible on iOS with a "real" mobile device, only with iPhone simulator in Firefox DevTools. Ignoring this bug for now as long as we cannot reproduce it on mobile.

**Comment by perlun** [comment]
> > A UI glitch in the show/hide TOC can be seen on mobile when dark mode is enabled (screenshot below)

Fixed in https://github.com/perlang-org/perlang/pull/176; upstreamed to darkerfx in https://github.com/perlun/darkerfx/pull/2

> This bug is btw also present in `darkfx`, so we could at the very least submit an issue to them about it at some point. Not yet fixed but I've started looking into it.

https://github.com/steffen-wilke/darkfx/issues/9

**Comment by perlun** [comment]
> There is currently no way to access the "theme switch" on mobile devices. This should be fixed. We can check how docs.microsoft.com has done it.

On second thought, this is not something that seems to be universally present. Here are my current findings:

* https://docs.microsoft.com does indeed have it. The theme switcher is located at the very bottom of the screen.
* https://docs.docker.com does _not_ allow the user to manually select the theme. Instead, it seems to rely on media-matching (much like we do it). Also, I realized mobile browsers typically allow the user to flip between night mode and daylight mode; at least Firefox has this feature on iOS. (Safari does not seem to have it.)

### Bottom line

I'm not in any way against adding this feature, I just don't want to spend my own precious time on it.  I'll file a follow-up issue about it, outlining the details if someone is interested in picking it up: #177.

I'll consider this issue completed now, closing issue.
